dali/operators/random/rng_base_gpu.cuh

```diff
@@ -28,33 +28,82 @@ namespace dali {
 
 namespace {
 
+template <bool value>
+using bool_const = std::integral_constant<bool, value>;
+
```

Comment:
Here I am dividing the implementation into four categories (cross product of is_noise_gen and is_per_channel)
can get different noise).